v2/local/keyword_search.go

```diff
@@ -1,2 +1,148 @@
 // Package local provides the features of the Local API.
 package local
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type SameName struct {
+	Region         []string `json:"region" xml:"region"`
+	Keyword        string   `json:"keyword" xml:"keyword"`
+	SelectedRegion string   `json:"selected_region" xml:"selected_region"`
+}
+
+type Place struct {
+	Id                string `json:"id" xml:"id"`
+	PlaceName         string `json:"place_name" xml:"place_name"`
+	CategoryName      string `json:"category_name" xml:"category_name"`
+	CategoryGroupCode string `json:"category_group_code" xml:"category_group_code"`
+	CategoryGroupName string `json:"category_group_name" xml:"category_group_name"`
+	Phone             string `json:"phone" xml:"phone"`
+	AddressName       string `json:"address_name" xml:"address_name"`
+	RoadAddressName   string `json:"road_address_name" xml:"road_address_name"`
+	X                 string `json:"x" xml:"x"`
+	Y                 string `json:"y" xml:"y"`
+	PlaceURL          string `json:"place_url" xml:"place_url"`
+	Distance          string `json:"distance" xml:"distance"`
+}
+
+type KeyWordSearchResult struct {
+	XMLName xml.Name `xml:"result"`
+	Meta    struct {
+		TotalCount    int  `json:"total_count" xml:"total_count"`
+		PageableCount int  `json:"pageable_count" xml:"pageable_count"`
+		IsEnd         bool `json:"is_end" xml:"is_end"`
+		SameName      `json:"same_name" xml:"same_name"`
+	} `json:"meta" xml:"meta"`
+	Documents []Place `json:"documents" xml:"documents"`
+}
+
+type KeyWordSearchInitializer struct {
+	Query             string
+	CategoryGroupCode string
+	Format            string
+	AuthKey           string
+	X                 string
+	Y                 string
+	Radius            int
+	Rect              string
+	Page              int
+	Size              int
+	Sort              string
+}
+
+func KeyWordSearch(query, x, y, category_group_code, rect string) *KeyWordSearchInitializer {
```

Comment:
The required parameter is only the `query`.
The rests would be better to be added using pointer receivers.